src/app/user-registration/user-registration.component.html

```diff
@@ -0,0 +1,90 @@
+<div class="mat_container">
+  <mat-card class="matcard_size">
+
+    <form (submit)="saveUser()" [formGroup]="signUpForm" class="height">
```

Comment:
Read more about reactive forms in angular. 
Add errors to each of the fields : https://medium.com/@vivek0208/create-form-using-angular-material-f2472dab1687

look for "<mat-error>" fields added to each of the fields in the form.

Reply:
Done.